Hot reload does not build multi module project in correct order

### Describe the bug

I have a multi module project, where I get exceptions on hot reload.
If I inly change one module at a time (and hot reload), no exception is thrown.

If i change multiple modules at the same time, I get the exception from below.
In module "service", I add a new method which I use in my rest resources in "web".
This new method is not found on hot reload.

I believe that the web module is compiled before service, otherwise the newly created method would be found.

### Expected behavior

Hot reload works flawlessly even when changing multiple modules at once.

### Actual behavior

Following exception:

```
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2022-01-25 07:09:35,054 WARN  [io.qua.config] (Quarkus Main Thread) Unrecognized configuration key "quarkus.datasource.db-kind" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo

2022-01-25 07:09:35,054 WARN  [io.qua.config] (Quarkus Main Thread) Unrecognized configuration key "quarkus.hibernate-orm.log.sql" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
2022-01-25 07:09:35,054 WARN  [io.qua.config] (Quarkus Main Thread) Unrecognized configuration key "quarkus.hibernate-orm.validate-in-dev-mode" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
2022-01-25 07:09:35,054 WARN  [io.qua.config] (Quarkus Main Thread) Unrecognized configuration key "quarkus.hibernate-orm.database.generation" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
2022-01-25 07:09:35,055 WARN  [io.qua.config] (Quarkus Main Thread) Unrecognized configuration key "quarkus.datasource.metrics.enabled" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
2022-01-25 07:09:35,737 INFO  [io.qua.sch.run.SimpleScheduler] (Quarkus Main Thread) No scheduled business methods found - Simple scheduler will not be started
2022-01-25 07:09:35,874 INFO  [io.quarkus] (Quarkus Main Thread) multi-module-hot-reload-web 1.0-SNAPSHOT on JVM (powered by Quarkus 2.6.1.Final) started in 4.437s. Listening on: http://localhost:8080
2022-01-25 07:09:35,874 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2022-01-25 07:09:35,874 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, config-yaml, hibernate-validator, jdbc-postgresql, rest-client, rest-client-jackson, resteasy-jackson, scheduler, smallrye-context-propagation, smallrye-health, smallrye-metrics, smallrye-openapi, swagger-ui, vertx]
2022-01-25 07:09:39,838 INFO  [io.qua.htt.access-log] ([DevConsole]vert.x-eventloop-thread-1) 127.0.0.1 - - 25/Jan/2022:07:09:39 +0000 "GET /q/dev HTTP/1.1" 302 -
2022-01-25 07:09:39,910 INFO  [io.qua.htt.access-log] ([DevConsole]vert.x-eventloop-thread-1) 127.0.0.1 - - 25/Jan/2022:07:09:39 +0000 "GET /q/dev/ HTTP/1.1" 200 51097

--
Compilation Failed:
warning: [options] system modules path not set in conjunction with -source 11
C:\workspaces\aviator\multi-module-hot-reload\multi-module-hot-reload-web\src\main\java\com\acme\rest\Resource.java:18: error: cannot find symbol
        return configuration.someSetting2();
                            ^
  symbol:   method someSetting2()
  location: variable configuration of type com.acme.config.ApplicationConfiguration
Tests paused
Press [r] to resume testing, [o] Toggle test output, [h] for more options>

```

### How to Reproduce?

1. Download the reproducer: 
[multi-module-hot-reload.zip](https://github.com/quarkusio/quarkus/files/7931394/multi-module-hot-reload.zip)

2. mvn clean compile quarkus:dev
3. open dev ui - this works
4. In com.acme.config.ApplicationConfiguration, remove the comments from `someSetting2`
5. In com.acme.rest.Resource, change the resource to `return configuration.someSetting()` instead of an empty string.
6. reload the dev ui
7. compilation error from above happens
8. Stop quarkus
9. mvn clean compile quarkus:dev
10. open dev ui - this works. The source itself is compileable by maven, but not by quarkus.

### Output of `uname -a` or `ver`

MSYS_NT-10.0-19043 NANB7NLNVP2 3.1.7-340.x86_64 2021-03-26 22:17 UTC x86_64 Msys

### Output of `java -version`

openjdk version "17" 2021-09-14 OpenJDK Runtime Environment Temurin-17+35 (build 17+35) OpenJDK 64-Bit Server VM Temurin-17+35 (build 17+35, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.6.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.3 (ff8e977a158738155dc465c6a97ffaf31982d739) Maven home: C:\eclipse\tools\apache-maven Java version: 17, vendor: Eclipse Adoptium, runtime: C:\eclipse\tools\java\17 Default locale: de_DE, platform encoding: Cp1252 OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

### Additional information

_No response_

--- a/core/deployment/src/main/java/io/quarkus/deployment/dev/DevModeContext.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/dev/DevModeContext.java
@@ -213,8 +213,8 @@ public class DevModeContext implements Serializable {
 
     public List<ModuleInfo> getAllModules() {
         List<ModuleInfo> ret = new ArrayList<>();
-        ret.add(applicationRoot);
         ret.addAll(additionalModules);
+        ret.add(applicationRoot);
         return ret;
     }
 

